File: UI/api/repository/models/architectural-design-recommendation.ts
/* tslint:disable */
/* eslint-disable */
import { ArchitecturalCategory } from './architectural-category';
import { ArchitecturalDesignSource } from './architectural-design-source';
import { QualityApproachAttributeEvaluation } from './quality-approach-attribute-evaluation';
import { QualityScore } from './quality-score';
import { QualitySublevelApproachAttributeEvaluation } from './quality-sublevel-approach-attribute-evaluation';
import { SystemPropertiesScore } from './system-properties-score';
export interface ArchitecturalDesignRecommendation {
  architecturalDesignId: number;
  architecturalDesignSource: ArchitecturalDesignSource;
  category: ArchitecturalCategory;
  identifier: string;
  matchesCount: number;
  qualityEvaluations?: null | Array<QualityApproachAttributeEvaluation>;
  qualityScore: QualityScore;
  qualitySublevelEvaluations?: null | Array<QualitySublevelApproachAttributeEvaluation>;
  suitabilityScore: number;
  systemPropertiesScore: SystemPropertiesScore;
  totalIncludeCount: number;
  totalScore: number;
  weightedScore: number;
}

File: ArchitectureRefactoringHelper/Repository/Models/Recommendation/ArchitecturalDesignRecommendation.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using Repository.Models.Database;
namespace Repository.Models.Recommendation
{
    public class ArchitecturalDesignRecommendation
    {
    [Required]
    [JsonPropertyName("architecturalDesignId")]
    public int ArchitecturalDesignId { get; set; }
    [Required]
    [JsonPropertyName("identifier")]
    public string Identifier { get; set; }

    [Required]
    [JsonPropertyName("architecturalDesignSource")]
    public ArchitecturalDesignSource ArchitecturalDesignSource { get; set; }


    [JsonPropertyName("qualityEvaluations")]
    public IList<ApproachAttributeEvaluation<Quality>> QualityEvaluations { get; set; }
    [JsonPropertyName("qualitySublevelEvaluations")]
    public IList<ApproachAttributeEvaluation<QualitySublevel>> QualitySublevelEvaluations { get; set; }

    [Required]
    [JsonPropertyName("suitabilityScore")]
    public int SuitabilityScore { get; set; }

    [Required]
    [JsonPropertyName("matchesCount")]
    public int MatchesCount { get; set; }

    [Required]
    [JsonPropertyName("totalIncludeCount")]
    public int TotalIncludeCount { get; set; }

    [Required]
    [JsonPropertyName("qualityScore")]
    public QualityScore? QualityScore { get; set; }

    [Required]
    [JsonPropertyName("systemPropertiesScore")]
    public SystemPropertiesScore? SystemPropertiesScore { get; set; }

    [Required]
    [JsonPropertyName("totalScore")]
    public int TotalScore { get; set; }

    [Required]
    [JsonPropertyName("weightedScore")]
    public int WeightedScore { get; set; }
        
    }
}